src/cryptoadvance/specter/templates/includes/coin_selection.html

```diff
@@ -0,0 +1,232 @@
+<template id="coin-selection">
+    <span class="toggle_coinselection" style="cursor: pointer;">{{ _("Coin selection") }} {% if show_advanced_settings %}&#9660;{% else %}&#9654;{% endif %}</span>
+    <br>
+    <div class="coinselection_div" style="margin: auto; max-width: 90%; display: none;">
+        <div class="errorMessagesDiv">
+            <message-box type="error" style="display: none;"></message-box>
+        </div>
+        <tx-table
+            class="txtable"
+            btc-unit="{{ specter.unit }}"
+            hide-sensitive-info="{{ specter.hide_sensitive_info | lower }}"
+            type="utxo" hide-columns="category time txid confirmations action" 
+            blockhash="true" hide-switcher="true">
+        </tx-table>
+    </div> 
+</template>
+
+
+<script type="module">
+    /**
+     * Custom element for showing the coin-selection-dialog
+       attributes:
+       * wallet-alias
+       * spendable-amount
+       Example:
+        <coin-selection  
+            id="coinselection-webcomponent"
+            wallet-alias="{{wallet.alias}}"
+            spendable-amount='{{ wallet.full_available_balance | btcamount }}'
+            is-liquid="{{specter.is_liquid}}"
+            selected-coins="{{ selected_coins }}">
+        </coin-selection>
+        Internally, a tx-table is used which is customizable in oder to be usefull here.
+        We're using here the asset_balance endpoint to get Liquid assets
+     */
+    class CoinSelection extends HTMLElement {
+        constructor() {
+            super();
+            var shadow = this.attachShadow({mode: 'open'});
+            var template_content = document.getElementById('coin-selection').content;
+            var clone = template_content.cloneNode(true);
+            
+            this.coinselectionDiv = clone.querySelector(".coinselection_div")
+            this.txtable = clone.querySelector(".txtable")
+            this.txtable.addEventListener("CustomSelected", (event) => {
+                this.updateLd()
+                this.fireChangeEvent();
+            } )
+
+            // ErrorMessages
+            this.errorMessagesDiv = clone.querySelector(".errorMessagesDiv");
+
+            // toggler
+            this.toggler = clone.querySelector(".toggle_coinselection")
+            this.toggler.addEventListener('click', (event) => {
+                this.toggleCoinselection()
+			});
+
+            this.createLighterDOMNodes()
+
+            // Attach the created element to the shadow dom
+            shadow.appendChild(clone);
+        }
+        /**
+         * Browser calls this method when the element is added to the document
+         * (can be called many times if an element is repeatedly added/removed)
+        */
+        connectedCallback() {
+            this.walletAlias = this.getAttribute('wallet-alias');
+            this.spendableAmount = this.getAttribute("spendable-amount") ?? 0
+            this.assetBalances = this.fetchAssetBalances()
```

Comment:
I think this would be better off as a json blob `data-` attribute so that we can eliminate the fetch.

Reply:
I passed it as a property. That's much easier as you don't have to fiddle with json and you don't have to implement `attributeChangedCallback` and `observedAttributes`.
Not entirely happy about the implementation, though. The assetBalances are not yet retrieved via the fetch-API in the wallet_sen.jinja but by templating in there.
But maybe we can change that in some other PR.